docusaurus/docs/reference/ha/operations.md

```diff
@@ -0,0 +1,96 @@
+---
+sidebar_label: Operations
+sidebar_position: 50
+---
+
+# Operating a Controller Cluster
+
+## Restoring from Backup
+
+To restore from a database snapshot, use the following CLI command:
+
+```
+ziti agent controller restore-from-db /path/to/backup.db
+```
+
+As this is an agent command, it must be run on the same machine as the controller. The path
+provided will be read by the controller process, not the CLI.
+
+The controller will apply the snapshot and then terminate. All controllers in the cluster will
+terminate and expect to be restarted. This is so in memory caches won't be out of sync with
+the database which has changed.
+
+## Snapshot Application and Restarts
+
+If a controller is out of communcation for a while, it may receive a snapshot to apply, rather
+than a stream of events.
+
+If a controller receives a snapshot to apply after starting up, it will apply the snapshot and then
+terminate. This assumes that there is a restart script which will bring the controller back up after
+it terminates.
+
+This should only happen if a controller is connected to the cluster and then gets disconnected for
+long enough that a snapshot is created while it's disconnected. Because applying a snapshot requires
+replacing the underlying controller bolt DB, the easiest way to do that is restart. That way we
+don't have to worry about replacing the bolt DB underneath a running system.
+
+## Events
+
+All events now contain a `event_src_id` to indicate which controller emitted them.
+
+There are some new events which are specific to clusters. See [Cluster Events](../events#cluster) 
+for more detail.
+
+## Metrics
+
+In an HA system, routers will send metrics to all controllers to which they are connected. There is
+a new `doNotPropagate` flag in the metrics message, which will be set to false until the router has
+successfully delivered the metrics message to a controller. The flag will then be set to true. So
+the first controller to get the metrics message is expected to deliver the metrics message to the
+events system for external integrators. The other controllers will have `doNotPropage` set to true,
+and will only use the metrics message internally, to update routing data.
+
+## Open Ports
+
+Controllers now establish connections with each other, for two purposes.
+
+1. Forwarding model updates to the leader, so they can be applied to the raft cluster
+2. raft communication
+
+Both kinds of traffic flow over the same connection.
+
+These connections do not require any extra open ports as we are using the control channel listener
```

Comment:
This implies the ALPN for controller-to-controller is something new, not `ziti-ctrl`. What's the new one? This is important for operators who probe their deployments to verify the correct certs are presented on certain ports, for example.

Reply:
It's still ziti-ctrl. The controller uses a channel framework header to distinguish between router and ctrl connections.